lighthouse-core/gather/gatherers/script-elements.js

```diff
@@ -26,6 +29,8 @@ function collectAllScriptElements() {
       async: script.async,
       defer: script.defer,
       source: /** @type {'head'|'body'} */ (script.closest('head') ? 'head' : 'body'),
+      // @ts-ignore - getNodePath put into scope via stringification
+      path: getNodePath(script),
```

Comment:
`path` might be a tad too generic on the `*Elements` artifacts for my taste, given it's basically a devtools-specific crazy node child index array

WDYT about...

`devtoolsNodePath`
`devtoolsChildIndexNodePath`
`nodePathForDevToolsRevealing`

any other suggestions welcome :)

Reply:
`nodeReferencePath`? I'd like to avoid tying it to DevTools, since other consumers could theoretically use it too.

So for now we'd use the new name here and keep `path` everywhere else?

Reply:
> since other consumers could theoretically use it too

😱 I hope not!

😆 If you can't tell, I'm not fond of this particular path approach, so I would prefer to limit its scope and push a more flexible strategy that can be resilient to reloads.

> So for now we'd use the new name here and keep path everywhere else?

yeah just thinking out loud here. I've proposed to only expose the `*Elements` artifacts as the public API for v5 so I'm thinking of them quite separately from the others at this point, which might not align with the rest of LH team yet :)

Reply:
>  I would prefer to limit its scope and push a more flexible strategy that can be resilient to reloads.

https://github.com/GoogleChrome/lighthouse/issues/2289#issuecomment-446606864 fyi

Reply:
Thanks @Hoten I'm glad we've already made some awesome improvements here :D 

To clarify, my main goal if we're exposing this to plugin consumers in `*Elements` would be to provide a path that has some value even when the node cannot be found, i.e. based on some sort of identifiers/classes/etc on the page and not just child indexes that don't provide much value to a human, some selector/path hybrid I guess.

Reply:
> 😱 I hope not!

Ok, `__legacyDevToolsPathDontUseEver` it is then 😂

> some selector/path hybrid I guess

Were you thinking of the `selector` property we show currently when the user hovers over a node in the report?

Axe already [aims to make that a unique selector](https://github.com/dequelabs/axe-core/blob/develop/lib/core/utils/get-selector.js#L336). [`getNodeSelector`](https://github.com/GoogleChrome/lighthouse/blob/b374ea409fc56fa1af7af8e9f3a5c6c1f96ee30d/lighthouse-core/lib/page-functions.js#L193) generates selectors based on the id/class name if available, but doesn't attempt to make that unique.

Seems like we don't need `path` at all if we can make `selector` unique.

Reply:
`__legacyDevToolsPathDontUseEver` 😍 love it! 😆 

I don't want to hold things up on this, but I'm just afraid of what we won't be able to take back once 5.0 hits :)

Anyone else core have thoughts on our public artifacts API? I think I might stir up this convo over on the issue...

Reply:
`devtoolsNodePath` sounds great to me, personally. Sounds uninvitingly opaque if you aren't devtools, you still know exactly what it is, and even if you do figure out how to use it, it sounds like the gears of devtools turning could change the format at any time.

Also fine with a leading underscore (but I vote we stay away from two :) and a strongly worded comment, but I think that's more than enough at that point.

Reply:
@paulirish if he has an opinion

Reply:
fine with `devtoolsNodePath`. seems cool.

Here's where the protocol uses it: https://vanilla.aslushnikov.com/?DOM.pushNodeByPathToFrontend